### PR TITLE
[REVIEW] Update to official libcu++ on Github

### PR DESCRIPTION
**Pull request by trxcllnt**
Update to libcu++ on Github.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by trxcllnt** [comment]
rerun tests

**Review by cwharris — CHANGES_REQUESTED** [review]
🔥

changes in the pr look good, but it’s not obvious what changes are made in the trxcllnt/libcudacxx clone. let’s get ops to clone libcudacxx, and then open a pr from the trxcllnt clone to the rapidsai clone so we can use the standard review process.

alternatively, we point at nvidia/libcudacxx and merge the type_trait/chrono backports there. thats preferable if libcudacxx team is cool with the changes.

File: thirdparty/CMakeLists.txt
Comment (trxcllnt) [reply thirdparty/CMakeLists.txt:48]: We're expecting the backports will be merged and upstreamed into libcxx, so this fork is temporary to see if it builds in CI.

**Review by harrism — APPROVED** [review]

**Comment by harrism** [comment]
Moving to  0.17

**Comment by jrhemstad** [comment]
libcu++ 1.4 should be release on Tuesday.

**Comment by ajschmidt8** [comment]
Removing `ops-codeowners` from the required reviews since it doesn't seem there are any file changes that we're responsible for. Feel free to add us back if necessary.

**Comment by harrism** [comment]
@kkraus14 and I discussed and agreed this is looking a bit risky for 0.17. We should get it done early in 0.18.

**Comment by harrism** [comment]
@trxcllnt please target branch-0.18 when ready.

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
retargeting `branch-0.17` so CI can run until the `gpuci/rapidsai:0.18` containers are published.

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Comment by harrism** [comment]
rerun tests

File: CHANGELOG.md
Comment (kkraus14) [inline CHANGELOG.md:294]: This needs to move to 0.18

File: cpp/src/rolling/rolling.cu
Comment (kkraus14) [inline cpp/src/rolling/rolling.cu:22]: Is there a reason these are quoted instead of bracketed now?

File: cpp/src/binaryop/binaryop.cpp
Comment (kkraus14) [inline cpp/src/binaryop/binaryop.cpp:25]: Quoted instead of bracketed now?

File: cpp/src/transform/transform.cpp
Comment (kkraus14) [inline cpp/src/transform/transform.cpp:21]: quoted instead of bracketed?

**Review by kkraus14 — CHANGES_REQUESTED** [review]

File: cpp/src/rolling/rolling.cu
Comment (trxcllnt) [reply cpp/src/rolling/rolling.cu:22]: Yes, IIRC it's because they're in the `src` dir and not the `include` dir. I recall we applied this convention for "private" headers that're under `src`, but lemme know if that's changed.

**Review by kkraus14 — APPROVED** [review]

File: cpp/src/rolling/rolling.cu
Comment (harrism) [reply cpp/src/rolling/rolling.cu:22]: We only use quotes when the path is in the same folder as the present file.  These used to be <>, which was correct, because while there is a src/rolling/jit folder, jit/launcher.h is in src/jit/, which is a different path.  

So this should be:

```suggestion
#include <jit/launcher.h>
#include <jit/parser.h>
#include <jit/type.h>

#include "jit/code/code.h"
#include "rolling_detail.hpp"
#include "rolling_jit_detail.hpp"
```

Also note that the latter three should be moved to the top of the file (as they were before), since they are the "nearest" includes, and we order from nearest to farthest.

(I just cleaned up the includes in nearly every file in the previous release.)

File: cpp/src/binaryop/binaryop.cpp
Comment (harrism) [reply cpp/src/binaryop/binaryop.cpp:25]:
```suggestion
#include "compiled/binary_ops.hpp" // move these three to the top, as they were
#include "jit/code/code.h"
#include "jit/util.hpp"

#include <jit/launcher.h>
#include <jit/parser.h>
#include <jit/type.h>
```

File: cpp/src/transform/transform.cpp
Comment (harrism) [reply cpp/src/transform/transform.cpp:21]:
```suggestion
#include "jit/code/code.h"

#include <jit/common_headers.hpp>
#include <jit/launcher.h>
#include <jit/parser.h>
#include <jit/type.h>
```

**Review by harrism — CHANGES_REQUESTED** [review]
Requesting changes for the #include fixes I suggested.

**Comment by harrism** [comment]
Make sure to remove all the strikethrough text from the description before merging.

**Review by cwharris — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Pull request merged**
